src/error.rs

```diff
@@ -213,6 +242,30 @@ impl Error {
     }
 }
 
+impl From<actix_http::error::Error> for Error {
+    fn from(error: actix_http::error::Error) -> Error {
+        Error::new(ErrorKind::ActixHttpError(error.to_string()), None)
+    }
+}
+
+impl From<serde_json::error::Error> for Error {
+    fn from(error: serde_json::error::Error) -> Error {
+        Error::new(ErrorKind::SerdeJsonError(error.to_string()), None)
+    }
+}
+
+impl From<r2d2::Error> for Error {
+    fn from(error: r2d2::Error) -> Error {
+        Error::new(ErrorKind::R2D2Error(error.to_string()), None)
+    }
+}
+
+impl From<String> for Error {
+    fn from(error: String) -> Error {
+        Error::new(ErrorKind::GenericError(error), None)
```

Comment:
I think GenericError will probably end up hiding too much information.  It's not necessarily going to be clear why it's being returned when it is.  It seems it will depend almost entirely on the error string in the originating error, which could be very short on useful information.

Reply:
Agreed. I got a bit carried away when doing the From impl's